src/oemof/solph/_plumbing.py

```diff
@@ -39,22 +44,55 @@ def sequence(iterable_or_scalar):
     >>> print(x)
     [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10]
 
+    >>> x = sequence({"len":9,"values":[0,1,2]})
+    >>> print(x)
+    [0, 0, 0, 1, 1, 1, 2, 2, 2]
+    >>> x[1]
+    0
+    >>> x[4]
+    1
+    >>> x[7]
+    2
+
+
     """
     if isinstance(iterable_or_scalar, abc.Iterable) and not isinstance(
         iterable_or_scalar, str
     ):
-        return iterable_or_scalar
+        if isinstance(iterable_or_scalar, abc.MutableMapping):
+            if (
+                iterable_or_scalar["len"] % len(iterable_or_scalar["values"])
+                != 0
+            ):
+                raise KeyError(
+                    "The length must be a multiple of the numer of the values!"
+                )
+            return _Sequence(
+                highest_index=iterable_or_scalar.get("len"),
+                period_values=iterable_or_scalar.get("values"),
+            )
+        else:
+            return iterable_or_scalar
     else:
         return _Sequence(default=iterable_or_scalar)
 
 
 class _Sequence(UserList):
-    """Emulates a list whose length is not known in advance.
+    """Emulates a list whose length is not known in advance if default
+    is passed. If period_values and highest_index are passed, the length
+    is known in advance and the sequence is periodic. The periods will have
+    the values passed in period_values and the lengths of the periods are equal
+    and sum up to the value of highest_index.
 
     Parameters
     ----------
     source:
-    default:
+    default: int or float or string
+        Default value for the sequence
+    period_values: iterable
```

Comment:
I personally would prefer `periodical_values`

Reply:
After some research, I will go for `periodic_values`